src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -1127,19 +1128,31 @@ def decrypt_key(cmd, client, algorithm, value, iv=None, tag=None, aad=None,
 
 
 def sign_key(cmd, client, algorithm, digest, name=None, version=None):
+    if '256' in algorithm:
```

Comment:
This should really be matching a list of precise allowed identifiers for the digest functions that are supported. Someone passing 'sha-224' for example will get a signature using sha-512, and that's not going to be what they want.

Reply:
All the options for algorithm are `ES256, ES256K, ES384, ES512, PS256, PS384, PS512, RS256, RS384, RS512`, so it's either `sha256`, `sha384` or `sha512`

Reply:
Ok, but this list is presumably checked elsewhere, and may get extended in the future, at which point this can become unsafe. Something like:

```
signing_algo_to_digest_algo = {
  "ES256": hashlib.sha256,
  ...
}

hash_func = signing_algo_to_digest_algo[algorithm]
```

would be clearer and safer.